### ISSUE TITLE
BUG: Replace fails after NaN in a Series of `string`

### ISSUE THREAD

**Issue by mhabets** [pandas-dev/pandas#59499]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
pd.options.future.infer_string = True
df = pd.Series(["toto", pd.NA, "titi", "tata"])
df.replace({"tata": "tutu"})
```

### Issue Description

Regex replace fails after the NaN value in a Series of `string`.
Result :
```
0    toto
1     NaN
2    titi
3    tata # Expecting: "tutu"
```

Note :
1. if NaN is before the value to be replaced like in `pd.Series(["toto", "titi", "tata", pd.NA])`, it works
2. `df.dropna().replace({"tata": "tutu"})` works as expected

### Expected Behavior

```
0    toto
1     NaN
2    titi
3    tutu
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : d9cdd2ee5a58015ef6f4d15c7226110c9aab8140
python                : 3.10.14.final.0
python-bits           : 64
OS                    : Windows
OS-release            : 10
Version               : 10.0.22631
machine               : AMD64
processor             : Intel64 Family 6 Model 186 Stepping 2, GenuineIntel
byteorder             : little
LC_ALL                : None
LANG                  : en
LOCALE                : English_United Kingdom.1252

pandas                : 2.2.2
numpy                 : 1.26.3
pytz                  : 2023.3.post1
dateutil              : 2.8.2
setuptools            : 69.0.3
pip                   : 23.3.2
Cython                : None
pytest                : None
hypothesis            : None
sphinx                : None
blosc                 : None
feather               : None
xlsxwriter            : None
lxml.etree            : None
html5lib              : None
pymysql               : None
psycopg2              : 2.9.9
jinja2                : 3.1.3
IPython               : 8.20.0
pandas_datareader     : None
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : None
bottleneck            : None
dataframe-api-compat  : None
fastparquet           : None
fsspec                : None
gcsfs                 : None
matplotlib            : 3.9.1
numba                 : None
numexpr               : None
odfpy                 : None
openpyxl              : 3.1.2
pandas_gbq            : None
pyarrow               : 16.1.0
pyreadstat            : None
python-calamine       : None
pyxlsb                : None
s3fs                  : None
scipy                 : 1.11.4
sqlalchemy            : 2.0.25
tables                : None
tabulate              : None
xarray                : None
xlrd                  : 2.0.1
zstandard             : None
tzdata                : 2023.4
qtpy                  : None
pyqt5                 : None

</details>

**Comment by asishm** [comment]
Thanks for the report. This seems to be a duplicate of #56599 which is fixed on main and will be available in v3.0

**Comment by mroeschke** [comment]
Closing as a duplicate
